### PR TITLE
[SPARK-24638][FOLLOW-UP][SQL] StringStartsWith UDP shall handle null values

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

`UserDefinedPredicate.keep(T)` shall handle `null` values because the **parquet-mr** library do invokes it maybe a null. This is the only way the filtering API can check if a null value fulfills the UDP.
It did not throw exception before the [ColumnIndex](https://issues.apache.org/jira/browse/PARQUET-1201) feature because the statistics and the dictionary filters invokes `keep(T)` only in some special scenarios (e.g. if all the values are null in the column chunk).
This pr handle null values for `StringStartsWith` UDP.

More details:
https://issues.apache.org/jira/browse/PARQUET-1488?focusedCommentId=16738074&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16738074

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @wangyum .
It would be great if you can elaborate the PR description a little bit more instead of give the link.

**Comment by wangyum** [comment]
Thanks @dongjoon-hyun Please review it later. I'm trying to add a test case.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala [hunk 1]
```diff
@@ -570,7 +570,7 @@ private[parquet] class ParquetFilters(
               }
 
               override def keep(value: Binary): Boolean = {
-                UTF8String.fromBytes(value.getBytes).startsWith(
+                value == null || UTF8String.fromBytes(value.getBytes).startsWith(
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala]: Does that actually affect any end-to-end job in Spark?

Comment (wangyum) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala]: @dongjoon-hyun @HyukjinKwon 
It seem does not affect Spark end-to-end work currently. 
It seem difficult to add a test case in the current Parquet version.
So are we merging this PR or closing it and fixing it in the future?

**Review by srowen — APPROVED** [review]

**Pull request closed without merging**